src/main/kotlin/org/wa/rceditor/api/RCManagerImpl.kt

```diff
@@ -0,0 +1,52 @@
+package org.wa.rceditor.api
+
+import io.reactivex.Completable
+import io.reactivex.Maybe
+import org.wa.rceditor.domain.RCManager
+import org.wycliffeassociates.resourcecontainer.ResourceContainer
+import org.wycliffeassociates.resourcecontainer.entity.Checking
+import org.wycliffeassociates.resourcecontainer.entity.DublinCore
+import org.wycliffeassociates.resourcecontainer.entity.Manifest
+import tornadofx.*
+
+class RCManagerImpl: RCManager {
+    override fun open(): Maybe<ResourceContainer> {
+        chooseDirectory("Open Resource Container").apply {
+            if (this != null) {
+                return try {
+                    Maybe.fromCallable {
+                        ResourceContainer.load(this)
+                    }
+                } catch (e: Exception) {
+                    println(e.message)
+                    Maybe.empty()
+                }
+            }
+        }
+        return Maybe.empty()
+    }
+
+    override fun create(): Maybe<ResourceContainer> {
+        chooseDirectory("Create Resource Container").apply {
+            if (this != null) {
+                return  try {
```

Comment:
Could you move this return in front of `chooseDirectory` to only have a single `return` call?

Reply:
chooseDirectory cannot be run on Schedulers.io, where I run ResourceContainer.load()

Reply:
Maybe you can suggest better solution?